SpiffWorkflow/spiff/specs/service_task.py

```diff
@@ -1,12 +1,20 @@
+import json
 from SpiffWorkflow.bpmn.specs.ServiceTask import ServiceTask
 from SpiffWorkflow.spiff.specs.spiff_task import SpiffBpmnTask
 
 class ServiceTask(SpiffBpmnTask, ServiceTask):
 
-    def __init__(self, wf_spec, name, operation_name, operation_params, **kwargs):
+    def __init__(self, wf_spec, name, operation_name, operation_params, result_variable, **kwargs):
         SpiffBpmnTask.__init__(self, wf_spec, name, **kwargs)
         self.operation_name = operation_name
         self.operation_params = operation_params
+        self.result_variable = result_variable
+
+    def _result_variable(self, task):
+        if self.result_variable is not None and len(self.result_variable) > 0:
+            return self.result_variable
+
+        return f'spiff__{task.task_spec.name}_result'
```

Comment:
do we need to normalize this in any way? can task_spec.name include spaces or hyphens?

Reply:
Good catch, looks like dashes can be in the name - I'll convert them to underscores.